### PR TITLE
{compiler} [GNU/4.9.3-2.25] Clang 3.7.0 (REVIEW)

### PR DESCRIPTION
**Pull request by wpoely86**
edit: requires #1977

### CODE REVIEW

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 13 out of 19 (11 easyconfigs in this PR)
Linux debian 8.1, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.9
See https://gist.github.com/5a82c23a780a5f44f718 for a full test report.

**Comment by boegel** [comment]
please resubmit test report?

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 19 out of 19 (11 easyconfigs in this PR)
Linux debian 8.1, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.9
See https://gist.github.com/60737b9d98c169622a19 for a full test report.

**Comment by wpoely86** [comment]
lgtm, ready to merge

**Comment by boegel** [comment]
It's your PR, of course it looks good to you ;)

**Comment by wpoely86** [comment]
Damn right. Nobody said I cannot comment on my own PR's :wink:

File: easybuild/easyconfigs/c/CMake/CMake-3.3.1-GNU-4.9.3-2.25.eb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+easyblock = 'ConfigureMake'
+
+name = 'CMake'
+version = '3.3.1'
```
Comment (boegel) [inline easybuild/easyconfigs/c/CMake/CMake-3.3.1-GNU-4.9.3-2.25.eb]: what's wrong with 3.3.2? ;)

Comment (wpoely86) [inline easybuild/easyconfigs/c/CMake/CMake-3.3.1-GNU-4.9.3-2.25.eb]: it was released after I made these.

File: easybuild/easyconfigs/c/CMake/CMake-3.3.2-GNU-4.9.3-2.25.eb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+easyblock = 'ConfigureMake'
+
+name = 'CMake'
+version = '3.3.2'
```
Comment (boegel) [inline easybuild/easyconfigs/c/CMake/CMake-3.3.2-GNU-4.9.3-2.25.eb]: pls remove, see #1991

File: easybuild/easyconfigs/n/ncurses/ncurses-6.0-GNU-4.9.3-2.25.eb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+easyblock = 'ConfigureMake'
+
+name = 'ncurses'
+version = '6.0'
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-6.0-GNU-4.9.3-2.25.eb]: pls remove, see #1991

File: easybuild/easyconfigs/c/CMake/CMake-3.3.2-GNU-4.9.3-2.25.eb (hunk 2 above)
Comment (wpoely86) [inline easybuild/easyconfigs/c/CMake/CMake-3.3.2-GNU-4.9.3-2.25.eb]: they will merge just fine.

File: easybuild/easyconfigs/n/ncurses/ncurses-6.0-GNU-4.9.3-2.25.eb (hunk 3 above)
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-6.0-GNU-4.9.3-2.25.eb]: or just resync with `develop` after #1977 is merged

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 11 out of 12 (11 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/83755b5978668a48687e for a full test report.

**Comment by boegel** [comment]
@wpoely86: download fail?

**Comment by boegel** [comment]
@wpoely86: please sync with develop (to get a clear diff here), and submit a working test report?

**Comment by wpoely86** [comment]
bizar. The url worked before and works on my end...

**Comment by boegel** [comment]
@wpoely86: maybe temporary issue with upstream, just try again?

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 11 out of 11 (3 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/9a6477569b29c752d2e5 for a full test report.

**Comment by wpoely86** [comment]
@boegel ready

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 4 (4 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/67ecbf49a24e47ee83df for a full test report.

**Comment by wpoely86** [comment]
@boegel it seems you used an old version of this PR?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (4 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/0d381eb13299b6f4af1a for a full test report.

**Comment by wpoely86** [comment]
huray

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/6555721fc7434ae8b74f for a full test report.

**Comment by wpoely86** [comment]
You broke it!
```
/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/CO7/haswell-ib/software/binutils/2.25-GCC-4.9.3-binutils-2.25/bin/ld:/tmp/vsc40023/easybuild_build/Clang/3.7.0/GNU-4.9.3-2.25/llvm.obj.1/projects/compiler-rt/lib/asan/clang_rt.asan-dynamic-x86_64.vers:1: syntax error in VERSION script
```
will have a look

**Comment by boegel** [comment]
I didn't break anything, I was being thorough by testing on two different systems. ;-)

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 10 out of 10 (4 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/369ef30ebb4af0c56de1 for a full test report.

**Comment by boegel** [comment]
@wpoely86: so, why did if work for you?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/d75ca00e75fc9010297a for a full test report.

**Comment by boegel** [comment]
Hmm, must have been a glitch in the matrix.

Going in, thanks @wpoely86!

**Comment by wpoely86** [comment]
@boegel If I have to guess, it's `-j24` that's the problem.

**Comment by boegel** [comment]
@wpoely86: well it was fine the 2nd time (which doesn't mean anything, I know)

**Comment by wpoely86** [comment]
Well yeah, probably the second time the order was as it should be.

So if anyone finds this in the future: use `maxparallel`.

**Comment by boegel** [comment]
Or we can add `maxparallel = 10` in the easyconfig?

Above 10 cores, I doubt you're going to see much more speedup.

**Comment by wpoely86** [comment]
yeah but the number 10 is completely arbitrary. In general I would suggest not to use 24 cores to build anything.

**Comment by boegel** [comment]
are you asking us to benchmark with 1..24? :p

**Comment by wpoely86** [comment]
Can't we use MPI or something and benchmark 1..256?

**Comment by wpoely86** [comment]
Maybe a great idea for a new EB part: find the optimal number of cores to build something :smile:

**Comment by boegel** [comment]
That sounds like a research paper to me. ;-)

**Pull request merged**